### ISSUE TITLE
Error serializing on homepage and 404 on slug pages in cms-prismic example

### ISSUE THREAD

**Issue by b-b0t** [zeit/next.js#12806]
# Bug report

## Describe the bug

Viewing of vanilla prismic example works fine on homepage but returns a 404 whenever navigating to slug pages. preview mode also returns this

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Clone cms-prismic example
2. Add env variables
3. yarn install && yarn dev / now dev
4. Navigate to any post

## Expected behavior

Preview not working
Live example allows viewing of blog post. (Currently password protected)

## Screenshots

<img width="961" alt="Screen Shot 2020-05-12 at 22 45 56 PM" src="https://user-images.githubusercontent.com/12721845/81743720-6b0f5080-94a2-11ea-88e2-7fa4116ce495.png">

<img width="400" alt="Screen Shot 2020-05-12 at 22 40 10 PM" src="https://user-images.githubusercontent.com/12721845/81743188-93e31600-94a1-11ea-85d5-d9ae0c81792c.png">

## System information

- OS: macOS 
- Browser: Chrome and Edge (Chromium)
- Version of Next.js: 9.2.3-canary.26 but also tried 9.4
- Version of Node.js: v10.16.3
- prismic-javascript: 2.7.1

## Additional context

When trying to integrate with latest packages from npm and Typescript there is an additional error related to the preview.

```
Error: Error serializing `.preview` returned from `getStaticProps` in "/".
Reason: `undefined` cannot be serialized as JSON. Please use `null` or omit this value all together.
```
Setting preview = null in `getStaticProps` removes this but the slug pages still return 404's.

**Comment by chibicode** [comment]
Hi @b-b0t, thanks for reporting! Just to confirm, did you follow all the steps on README: https://github.com/zeit/next.js/blob/canary/examples/cms-prismic/README.md ? Asking because your “To Reproduce” steps didn’t mention anything about setting up Prismic.

Can you try putting `console.log` somewhere on `fetchAPI` and check to see where things are failing?
https://github.com/zeit/next.js/blob/canary/examples/cms-prismic/lib/api.js

**Comment by b-b0t** [comment]
thanks @chibicode.

followed the steps and used the json object to generate the prismic custom types just in case.
sorry if anything was unclear, stated that the homepage works fine (posts pull through) and I added the env variables which include my API key from prismic.

logging shows a 200 on the prismic api. I did notice the following though:
`previewSession` is being used but this has been deprecated in favour of `getPreviewResolver`
https://github.com/zeit/next.js/blob/canary/examples/cms-prismic/pages/api/preview.js

when replacing with new method getting the same error, but maybe I'm missing something

**Comment by b-b0t** [comment]
closed without resolution 🤷🏼‍♂️
can't confirm because https://next-blog-prismic.now.sh/ is locked
oh well

**Comment by ismamz** [comment]
@b-b0t Check that the `NEXT_EXAMPLE_CMS_PRISMIC_REPOSITORY_LOCALE` matches with your repository's language on Prismic (/settings/multi-languages/).

**Comment by chibicode** [comment]
@b-b0t Sorry about it being locked, we unlocked it: https://next-blog-prismic.now.sh/posts/from-server-side-rendering-to-static-generation

Could you look into `getAllPostsWithSlug`? Is `data` empty? https://github.com/zeit/next.js/blob/b41f9baaa413d5dac29faf107663214c0923c8bd/examples/cms-prismic/lib/api.js#L42-L57

```js
  const data = await fetchAPI(`
    {
      allPosts {
        edges {
          node {
            _meta {
              uid
            }
          }
        }
      }
    }
  `)
```

**Comment by b-b0t** [comment]
@chibicode I have since switched to the Sanity example and it's working perfectly (what a great platform). However I did check and the slugs were being returned.

@ismamz was correct, it was defaulting to `en-us` but my Prismic settings geolocated me to `en-gb`. The strange thing is the posts on the homepage were fine, but only when using `yarn dev` not with `now dev`. Submitted PR so this doesn't happen to anyone else. #12862 

As the examples evolve might want to put some fail safes in place as this was breaking in production too.

Thanks for the help guys. Still gonna stick with Sanity.
